designs/developer/health checks/health checks-45.md

```diff
@@ -0,0 +1,55 @@
+![add-hc-2c](../health checks/img/add-hc-2c.png)
+
+
+## Edit Health Checks
+Users can add or edit health checks using the actions dropdown menu.  If there are no health checks configured the action will be “Add Health Checks”. 
+![hc-sidepanel-actionsadd](../health checks/img/hc-sidepanel-actionsadd.png)
+
+If at least one health check is configured, the action will be “Edit Health Checks”.
+![hc-sidepanel-actionsedit](../health checks/img/hc-sidepanel-actionsedit.png)
+
+If there are no health checks configured the title of the form will be “Add Health Checks” and the actions at the bottom will be “Add” and “Cancel”. If only one container exists, that container name will be listed at the top after “Container”.
+![add-hc-1c](../health checks/img/add-hc-1c.png)
+
+If more than one container exists in the deployment/deployment config, the container name will be listed in a dropdown after “Container”. 
+![add-hc-2c](../health checks/img/add-hc-2c.png)
+
+When there is more than one container, the dropdown will be disabled once the user begins to add health checks.  Depending on which health check the user chooses to add or edit, the corresponding part of this form will expand. The user must then click save or cancel to the entire page. Refer to the [Health Checks Add Form](developer/add-45/sections/Adv-Health_Checks.md) for the expanded forms and interactions.
+![add-hc-expanded](../health checks/img/add-hc-expanded.png)
+
+If there is at least one health check configured the title of the form will be “Edit Health Checks” and the actions at the bottom will be “Save” and “Cancel”. If only one container exists, that container name will be listed at the top after “Container”.
+![edit-hc-1c](../health checks/img/edit-hc-1c.png)
+
+If more than one container exists in the deployment/deployment config, the container name will be listed in a dropdown after “Container”. 
+![edit-hc-2c](../health checks/img/edit-hc-2c.png)
+
+When there is more than one container, the dropdown will be disabled once the user begins to add health checks.  Depending on which health check the user chooses to add or edit, the corresponding part of this form will expand. The user must then click save or cancel to the entire page. Refer to the [Health Checks Add Form](developer/add-45/sections/Adv-Health_Checks.md)for the expanded forms and interactions.
+![edit-hc-expanded](../health checks/img/edit-hc-expanded.png)
+
+## Health Check Erros
+In 4.5, Health Check Errors will be visible in the Monitoring tab of the Topology view. Health Check violations should be warning events, thus will be shown under the Events (Warning) section of the Monitoring tab in the side panel. In the future, when the Notification Drawer is available for Developers, we would expect this to be shown in the drawer as well.
```

Comment:
Do these events exist today? I think events are usually are triggered by something happening, and aren't stored for very long. If they do exist, they would also appear in the Project's Overview and Events tabs.

That makes me wonder if we should explore exposing whether or not Health Checks are configured within the Project Overview's Status card, either in the bar at the top or as a status message with a popover that shows the offending Deployments. That might make it easier for developers to find the Deployments that still need health checks within the particular project/s they care about. [Here's some design documentation](http://openshift.github.io/openshift-origin-design/designs/administrator/4.3/dashboards/#status-card) on the Status card for reference.

Reply:
@andybraren Something else we can have Brigid respond to when she gets back. I believe these events persist to some degree. There is a scenario where the event can get fired and persists but then probe resolves/changes.